fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexOfFunctionKeyExpression.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * IndexOfFunctionKeyExpression.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.ObjectPlanHash;
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.metadata.expressions.FunctionKeyExpression;
+import com.apple.foundationdb.record.metadata.expressions.KeyExpression;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecord;
+import com.apple.foundationdb.record.util.HashUtils;
+import com.google.auto.service.AutoService;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Function key expression that computes the index of an item within a list. This key expression takes
+ * two arguments:
+ *
+ * <ol>
+ *     <li>A list of items.</li>
+ *     <li>A scalar object.</li>
+ * </ol>
+ *
+ * <p>
+ * It returns a single column, which will contain the (zero-indexed) position of the scalar object
+ * within the list. For example, on a message like:
```

Comment:
The _first occurrence_ of the object in the list, right? Also, `null` if not found?